### ISSUE TITLE
City Strength with Garrison is not working properly

### ISSUE THREAD

**Issue by Aqinu** [LoneGazebo/Community-Patch-DLL#6166]
_1. Mod version (i.e Date - 4/23): 1-11

_2. Mod list (if using Vox Populi only, leave blank):_

_3. Error description: 

A city with or without wall works correctly (same strength both sides, 30 damage inflicted on each sides), but with garrison inside the city it's not longer the case (even though the strength is the same).

When the unit it is attacked by the city it seems that works correctly in every case.

_4. Steps to reproduce (optional):_

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. Make sure you have enabled logging before experiencing an error! Go here to find out how: http://forums.civfanatics.com/showthread.php?t=487482):

2. Save game (always attach a save that was made a turn before the error; located at My Documents/My Games/Sid Meier's Civilization 5/ModdedSaves):

3. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

4. Screenshots (optional):

**Here same strength same value inflicted correctly (30)**
![a](https://user-images.githubusercontent.com/49090452/73134503-d80bf100-403f-11ea-9c38-2c0fb68f1dad.png)

**Here with garrison, same strength it not longer correct (16/30)**
![b](https://user-images.githubusercontent.com/49090452/73134519-f40f9280-403f-11ea-8b76-44f821ffb6b0.png)

**Here with walls everything it's fine**
![c](https://user-images.githubusercontent.com/49090452/73134523-fb36a080-403f-11ea-8ff4-ddd56bcc29bb.png)

**Here with garrison again not correct**
![d](https://user-images.githubusercontent.com/49090452/73134530-05f13580-4040-11ea-8315-fb62024d027f.png)

**Comment by Stalker012345** [comment]
The pics are showing the melee unit attacking the city. When the garrison is placed inside it, the city's CS goes up, and therefore the melee units damage goes down (which happens anytime the defender's CS goes up).

I'm not seeing the issue personally.

**Comment by Aqinu** [comment]
@Stalker0123451
1. no garrison - with same strength, 30 (unit) 30 (city) damage inflicted
2. with garrison - with same strength,  20 (unit) 30 (city) damage inflicted

No issue? With our without garrison if it is the same strength (both sides) should inflict same damage...

**Comment by Stalker012345** [comment]
In the first picture, the City's CS is 9. In the second its 11. So its not the same strength, the garrison is in fact raising the city's CS.

**Comment by Aqinu** [comment]
@Stalker012345 I said the **same strength in the same screenshot** not between screenshots, i put multiple to get the point.

First screenshot approx. same strength,(the 0.6, as much as close as i get, not much difference), let's say those were equal unit (9 strength), city (9 strength), both were inflicting 30 damage.
Second screenshot exactly the same strength (11 unit, 11 city), unit inflicts 20 (instead of 30), city inflicts 30, what don't you see, if the city has garrison has other rules of calculating that inflicting damage...

**Comment by UndeadDevel** [comment]
What's also weird is that the HP bars in the combat preview are depleted to pretty much the same degree for the attacking unit in all cases...even though in the screenshots with ~16 damage they should be shown at lower levels of depletion...so maybe it's just a UI issue? Did you test what damage was actually being done when the attack was carried out?

**Comment by ilteroi** [comment]
i have an idea: let's remove those city attacks for good ...

sorry, had to say it, my head made me do it.

**Comment by UndeadDevel** [comment]
Well this is about attacks on Cities, though, not City Attacks, but yes, let's remove all the clutter from this game, like Cities and terrain, for example. I think if we made Duel mode obligatory, made the map square with only Grassland tiles and gave the two players a row of Warriors each with the victory condition being elimination of all enemy units and introducing an effect that upon reaching the other side with one of your Warriors, it will turn into a GDR, then we will finally achieve perfect balance 😃

**Comment by LoneGazebo** [comment]
Okay, so I don't see any issues here...what's the problem, exactly? The code looks fine.

**Comment by Aqinu** [comment]
@LoneGazebo are you serious?
So there's no issue if a city has garrison and the "city strength acts as a unit in offense/defense" no longer applies in this instance...

**Comment by LoneGazebo** [comment]
I really don't understand what you're trying to say.

**Comment by Aqinu** [comment]
@LoneGazebo Oh man, that bad am I explaining? Let me try again.

AFAIK when a unit has for example 10 strength and it's attacking another unit which also has 10 strength (the strength shown in the combat panel obviously) both of the units inflict 30 damage to one another (of course if the attacking unit is ranged, only inflicts damage, doesn't take damage).

Here change from the current version (1/11), about Cities
> They behave identically to real units, where their strength is equal to their defense and their damage output

So let's say that a melee unit (**15 strength** in combat panel) attacks a **City WITHOUT Garrison** (no units in the city, also **15 strength** in combat panel shown), **both of them inflict to one another 30 damage** (also shown in the combat panel, **which is expected**)

BUT if a melee unit (still **15 strength** in combat panel) attack a **City WITH Garriso**n (a unit is in the city, also the **total strength,including the strength from the unit**, ends up still **15 strength** for this city, **the strength same as above**), the melee unit only inflicts **20 damage** (which should be still 30, there are the same numbers here and in the previous example), and the city inflicts **30 damage** (which is as above, it's ok)

As far as I see it when a unit stays in the city, the city strength works in combat as it did before this current change, here should be made the changes neccessary to work with the current system, so that city strength with a unit behaves as a normal unit in combat.

Please tell me what don't you understand, Please tell me what you do understand, cause I don't know how can I explained it better than this...

**Comment by ilteroi** [comment]
the damage from the attack will be split between the city and the garrison. iirc, the preview only shows the city's share.

**Comment by Aqinu** [comment]
>the damage from the attack will be split between the city and the garrison. iirc, the preview only shows the city's share.

That I didn't know, I though that
>They behave identically to real units, where their strength is equal to their defense and their damage output

includes this scenario also (city with garrison, and the damage it's not share with the unit but is not the case), but apparently no, so that statement it's not universal... (cause you do a lot less damage to the city if it is any garrison present)

**Comment by Aqinu** [comment]
If this works as intended it's fine I just wanted to point out that, the statement it's not 100% correct

>They behave identically to real units, where their strength is equal to their defense and their damage output

**Comment by LoneGazebo** [comment]
@ilteroi possible to have final damage calc here show total damage to city and garrison, just for UI? Seems like the better choice.

**Comment by Aqinu** [comment]
@LoneGazebo 

Personally I don't like the idea, when i hit the city i want to know how much damage I make and how fast will fall based on the city health, why not make the damage for the garrison unit separated (a percentage of the damage inflicted on the city), so that the city strength will really behave as a normal unit.

**Comment by LoneGazebo** [comment]
>why not make the damage for the garrison unit separated (a percentage of the damage inflicted on the city)
you'd be double-dipping on damage

**Comment by Aqinu** [comment]
I know but if you really want to have "cities behave as normal units" you have to do something like this, than raise the CS of the city cause you will do more damage if it's not shared with the garrison.

I don't see such an issue that can't be tweaked.
